### PR TITLE
refactor: publish core package

### PR DESCRIPTION
**Pull request by cvarjao**
# Summary of Changes

- remove `packages/core` in favour of `packages/legacy/core`
- publish `packages/legacy/core` as `@hyperledger/aries-bifold-core`
- move verifier code to its own package: `@hyperledger/aries-bifold-verifier`
- standardize building/compiling process around [react-native-builder-bob](https://github.com/callstack/react-native-builder-bob)
- standardize/align/share typescript configuration file

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [ ] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [ ] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [ ] Updated documentation as needed for changed code and new or modified features;
- [ ] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Review by bryce-mcmath — COMMENTED** [review]
I don't see any issues, but are there no changes required in `workflows/publish.yaml` and `workflows/main.yaml`?

File: packages/legacy/app/package.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
     "android": "react-native run-android",
     "gradle": "./android/gradlew --project-dir=./android --no-daemon",
     "gradle:write-locks": "cd android && ./gradlew app:dependencies --write-locks",
-    "postinstall": "patch-package && react-native setup-ios-permissions && pod-install --non-interactive",
+    "prepare": "patch-package",
```
Comment (jleach) [inline packages/legacy/app/package.json:10]: Is this still needed? I think we use yarn patches now. No?

File: tsconfig-base.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+  "compilerOptions": {
+    "target": "esnext" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */,
```
Comment (jleach) [inline tsconfig-base.json:3]: Ok to do comments in JSON like this?

**Review by jleach — APPROVED** [review]
Two comments / observations. No blockers. LGTM.

**Pull request merged**